infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -343,6 +376,65 @@ public void prefixTrim(long address) {
         logMetadata.prefixTrim(address);
     }
 
+    @Override
+    public void suffixTrim() {
+
+        // Have a global tail copy is necessary, since degrade to reset is dependent on it
+        long globalTailCopy = logMetadata.getGlobalTail();
+
+        try {
+            long committedTail = dataStore.getCommittedTail();
+
```

Comment:
I think its also a good idea to check that the committed tail is initialized. Because otherwise it will throw an isTrimmed exception, since the committed tail is initialized to -1 and the starting address is initialized to 0.